### PR TITLE
stm32: l0: timer/lptimer: doc+

### PR DESCRIPTION
**Pull request by grevaillot**
So, i had that for a while in a tree here : improve doc for l0 lptim (in timer.h), and add MASK defines - let's push it before someone else does it :)

### CODE REVIEW

File: include/libopencm3/stm32/l0/timer.h [hunk 1]
```diff
@@ -138,31 +138,53 @@ LGPL License Terms @ref lgpl_license
 #define LPTIM_IER_UPIE			(1 << 5)
 #define LPTIM_IER_DOWNIE		(1 << 6)
 
+/* --- LPTIM_CFGR values ----------------------------------------------------- */
```
Comment (karlp) [inline include/libopencm3/stm32/l0/timer.h]: If you're going to add a "blah values" section, you might as well make it a doxygen group.  We've done that in other places, so there's a group for all the values of each register.  Just this ----- line doesn't really do much.

File: include/libopencm3/stm32/l0/timer.h [hunk 2]
```diff
@@ -138,31 +138,53 @@ LGPL License Terms @ref lgpl_license
...
+
+/* CKSEL: Select internal (0) or external clock source (1) */
```
Comment (karlp) [inline include/libopencm3/stm32/l0/timer.h]: you need a /** to make doxygen pick it up

File: include/libopencm3/stm32/l0/timer.h [hunk 3]
```diff
@@ -138,31 +138,53 @@ LGPL License Terms @ref lgpl_license
...
 #define LPTIM_CFGR_CKSEL		(1 << 0)
 
 #define LPTIM_CFGR_CKPOL_SHIFT		1
+#define LPTIM_CFGR_CKPOL_MASK		0x03
 #define LPTIM_CFGR_CKPOL		(3 << LPTIM_CFGR_CKPOL_SHIFT)
+/** @defgroup lptim_cfgr_ckpol LPTIM_CFGR CKPOL Clock Polarity
+* @ingroup timer_defines
```
Comment (karlp) [inline include/libopencm3/stm32/l0/timer.h]: You can skip the @ingroup when it's naturally inside that group.  You only need to explicitly do an @ingroup if you want to put it "elsewhere"

File: include/libopencm3/stm32/l0/timer.h [hunk 4]
```diff
@@ -182,30 +209,47 @@ LGPL License Terms @ref lgpl_license
...
+/* @def LPTIM_CFGR_TRIGEN LPTIM_CFGR TRIGEN Trigger enable and polarity
+* @ingroup timer_defines
+@{*/
 #define LPTIM_CFGR_TRIGEN_SW		(0 << LPTIM_CFGR_TRIGEN_SHIFT)
 #define LPTIM_CFGR_TRIGEN_RISING	(1 << LPTIM_CFGR_TRIGEN_SHIFT)
 #define LPTIM_CFGR_TRIGEN_FALLING	(2 << LPTIM_CFGR_TRIGEN_SHIFT)
 #define LPTIM_CFGR_TRIGEN_BOTH		(3 << LPTIM_CFGR_TRIGEN_SHIFT)
+/**@}*/
 
+/* TIMOUT: Timeout enable */
```
Comment (karlp) [inline include/libopencm3/stm32/l0/timer.h]: Same for all of these, they need the second **

File: include/libopencm3/stm32/l0/timer.h (hunk 1 above)
Comment (grevaillot) [reply include/libopencm3/stm32/l0/timer.h]: hmm, do you have some good doc example/guideline to follow to keep things consistent ? I aligned with timer_common_all.h :)

File: include/libopencm3/stm32/l0/timer.h (hunk 2 above)
Comment (grevaillot) [reply include/libopencm3/stm32/l0/timer.h]: ahhh, indeed. thanks

File: include/libopencm3/stm32/l0/timer.h (hunk 3 above)
Comment (grevaillot) [reply include/libopencm3/stm32/l0/timer.h]: hmm, that does not seem to work here without explicit ingroup :( - not sure how/why right now, but same thing is done in timer_common_all.h. I'll have a look.

**Comment by grevaillot** [comment]
ok, i fixed the defines comment to have them picked up and grouped registers to have something nicer, but i still need to add some ingroup @timer_defines to have them in the right place.

File: include/libopencm3/stm32/l0/timer.h [hunk 5]
```diff
@@ -182,30 +218,52 @@ LGPL License Terms @ref lgpl_license
 /* 5 is reserved */
 #define LPTIM_CFGR_TRIGSEL_EXT_TRIG6	(6 << LPTIM_CFGR_TRIGSEL_SHIFT)
 #define LPTIM_CFGR_TRIGSEL_EXT_TRIG7	(7 << LPTIM_CFGR_TRIGSEL_SHIFT)
+/**@}*/
 
 #define LPTIM_CFGR_TRIGEN_SHIFT		17
+#define LPTIM_CFGR_TRIGSEL_MASK		0x07
 #define LPTIM_CFGR_TRIGEN		(3 << LPTIM_CFGR_TRIGEN_SHIFT)
+/* @defgroup LPTIM_CFGR_TRIGEN LPTIM_CFGR TRIGEN Trigger enable and polarity
```
Comment (karlp) [inline include/libopencm3/stm32/l0/timer.h]: hah, this is what was giving me troubles with groupings, it's missing the second *  I've fixed it locally, no need to repush anything.

**Comment by karlp** [comment]
this has landed already, I fixed up the extr abits locally when I merged it. thanks!

**Pull request closed without merging**